ogr/ogrsf_frmts/hana/ogrhanautils.cpp

```diff
@@ -0,0 +1,227 @@
+    va_end( args );
+    return CPLString(buffer.data(), size);
+}
+
+const char* SkipLeadingSpaces(const char* value)
+{
+    while (*value == ' ')
+        value++;
+    return value;
+}
+
+CPLString JoinStrings(
+    const std::vector<CPLString>& strs,
+    const char* delimiter,
+    CPLString (*decorator)(const CPLString& str))
+{
+    CPLString ret;
+    for (std::size_t i = 0; i < strs.size(); ++i)
+    {
+        ret += ((decorator != nullptr) ? decorator(strs[i]) : strs[i]);
+        if (i != strs.size() - 1)
+            ret += delimiter;
+    }
+    return ret;
+}
+
+std::vector<CPLString> SplitStrings(const char* str, const char* delimiter)
+{
+    std::vector<CPLString> ret;
+    if (str != nullptr)
+    {
+        char** items = CSLTokenizeString2(str, delimiter, CSLT_HONOURSTRINGS);
+        for (int i = 0; items[i] != nullptr; ++i)
+        {
+            CPLString item(items[i]);
+            ret.push_back(item.Trim());
+        }
+
+        CSLDestroy(items);
+    }
+
+    return ret;
+}
+
+CPLString GetFullTableName(
+    const CPLString& schemaName, const CPLString& tableName)
+{
+    if (schemaName.empty())
+        return tableName;
+    return schemaName + "." + tableName;
+}
+
+CPLString GetFullTableNameQuoted(
+    const CPLString& schemaName, const CPLString& tableName)
+{
+    if (schemaName.empty())
+        return QuotedIdentifier(tableName);
+    return QuotedIdentifier(schemaName) + "." + QuotedIdentifier(tableName);
+}
+
+CPLString GetFullColumnNameQuoted(
+    const CPLString& schemaName,
+    const CPLString& tableName,
+    const CPLString& columnName)
+{
+    return GetFullTableNameQuoted(schemaName, tableName) + "."
+           + QuotedIdentifier(columnName);
+}
+
+CPLString LaunderName(const char* name)
+{
+    if (name == nullptr)
+        return nullptr;
+
+    CPLString newName(name);
+    for (std::size_t i = 0; newName[i] != '\0'; ++i)
+    {
+        char c = static_cast<char>(toupper(newName[i]));
+        if (c == '-' || c == '#')
+            c = '_';
+        newName[i] = c;
+    }
+    return newName;
+}
+
+CPLString Literal(const CPLString& value)
+{
+    CPLString ret("'");
+    char* tmp = CPLEscapeString(value, -1, CPLES_SQL);
+    ret += tmp;
+    CPLFree(tmp);
+    ret += "'";
+    return ret;
+}
+
+CPLString QuotedIdentifier(const CPLString& value)
+{
+    return "\"" + value + "\"";
+}
+
+bool IsGeometryTypeSupported(OGRwkbGeometryType wkbType)
+{
+    switch (wkbFlatten(wkbType))
+    {
+    case OGRwkbGeometryType::wkbPoint:
+    case OGRwkbGeometryType::wkbLineString:
+    case OGRwkbGeometryType::wkbPolygon:
+    case OGRwkbGeometryType::wkbMultiPoint:
+    case OGRwkbGeometryType::wkbMultiLineString:
+    case OGRwkbGeometryType::wkbMultiPolygon:
+    case OGRwkbGeometryType::wkbGeometryCollection:
+    case OGRwkbGeometryType::wkbCircularString:
+        return true;
+    default:
+        switch (wkbType)
+        {
+        case OGRwkbGeometryType::wkbPoint25D:
```

Comment:
I don't think those cases can be hit as wkbFlatten(OGRwkbGeometryType::wkbPoint25D) == OGRwkbGeometryType::wkbPoint matched above